var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -79,24 +79,96 @@ class Petsc(Package, CudaPackage, ROCmPackage):
     version("3.14.2", sha256="87a04fd05cac20a2ec47094b7d18b96e0651257d8c768ced2ef7db270ecfb9cb")
     version("3.14.1", sha256="0b4681165a9af96594c794b97ac6993452ec902726679f6b50bb450f89d230ed")
     version("3.14.0", sha256="a8f9caba03e0d57d8452c08505cf96be5f6949adaa266e819382162c03ddb9c5")
-    version("3.13.6", sha256="67ca2cf3040d08fdc51d27f660ea3157732b24c2f47aae1b19d63f62a39842c2")
-    version("3.13.5", sha256="10fc542dab961c8b17db35ad3a208cb184c237fc84e183817e38e6c7ab4b8732")
-    version("3.13.4", sha256="8d470cba1ceb9638694550134a2f23aac85ed7249cb74992581210597d978b94")
-    version("3.13.3", sha256="dc744895ee6b9c4491ff817bef0d3abd680c5e3c25e601be44240ce65ab4f337")
-    version("3.13.2", sha256="6083422a7c5b8e89e5e4ccf64acade9bf8ab70245e25bca3a3da03caf74602f1")
-    version("3.13.1", sha256="74a895e44e2ff1146838aaccb7613e7626d99e0eed64ca032c87c72d084efac3")
-    version("3.13.0", sha256="f0ea543a54145c5d1387e25b121c3fd1b1ca834032c5a33f6f1d929e95bdf0e5")
-    version("3.12.5", sha256="d676eb67e79314d6cca6422d7c477d2b192c830b89d5edc6b46934f7453bcfc0")
-    version("3.12.4", sha256="56a941130da93bbacb3cfa74dcacea1e3cd8e36a0341f9ced09977b1457084c3")
-    version("3.12.3", sha256="91f77d7b0f54056f085b9e27938922db3d9bb1734a2e2a6d26f43d3e6c0cf631")
-    version("3.12.2", sha256="d874b2e198c4cb73551c2eca1d2c5d27da710be4d00517adb8f9eb3d6d0375e8")
-    version("3.12.1", sha256="b72d895d0f4a79acb13ebc782b47b26d10d4e5706d399f533afcd5b3dba13737")
-    version("3.12.0", sha256="ba9ecf69783c7ebf05bd1c91dd1d4b38bf09b7a2d5f9a774aa6bb46deff7cb14")
-    version("3.11.4", sha256="319cb5a875a692a67fe5b1b90009ba8f182e21921ae645d38106544aff20c3c1")
-    version("3.11.3", sha256="199ad9650a9f58603b49e7fff7cd003ceb03aa231e5d37d0bf0496c6348eca81")
-    version("3.11.2", sha256="4d244dd7d1565d6534e776445fcf6977a6ee2a8bb2be4a36ac1e0fc1f9ad9cfa")
-    version("3.11.1", sha256="cb627f99f7ce1540ebbbf338189f89a5f1ecf3ab3b5b0e357f9e46c209f1fb23")
-    version("3.11.0", sha256="b3bed2a9263193c84138052a1b92d47299c3490dd24d1d0bf79fb884e71e678a")
+    version(
+        "3.13.6",
+        sha256="67ca2cf3040d08fdc51d27f660ea3157732b24c2f47aae1b19d63f62a39842c2",
+        deprecated=True,
```

Comment:
Due to dependencies - the same versions in py-petsc4py, slepc, and py-slepc4py should also be deprecated.

cc: @joseeroman

Reply:
Ok - I can create a subsequent PR for these pkgs.

> The important thing is that installing any of these deprecated versions (including Python 3.8 itself) will raise a warning telling users that this version is deprecated and they should update

```console
balay@pj01:~/spack$ ./bin/spack spec slepc@3.13.0
==> Error: failed to concretize `slepc@3.13.0` for the following reasons:
     1. Cannot satisfy 'petsc@3.13'
     2. Cannot satisfy 'petsc@3.13'
        required because slepc depends on petsc@3.13 when @3.13 
          required because slepc@3.13.0 requested explicitly 
balay@pj01:~/spack$ ./bin/spack spec --deprecated slepc@3.13.0
==> Warning: using "petsc@3.13.6" which is a deprecated version
==> Warning: using "python@3.8.19" which is a deprecated version
==> Warning: using "cmake@3.28.2" which is a deprecated version
 -   slepc@3.13.0%gcc@14.2.1+arpack~blopex~cuda~hpddm~rocm build_system=generic arch=linux-fedora40-skylake
 -       ^arpack-ng@3.9.0%gcc@14.2.1~icb~ipo+mpi+shared build_system=cmake build_type=Release generator=make arch=linux-fedora40-skylake
```
Hm - spack is treating "deprecated" version as "removed" by default - so it won't be obvious to most users that the trigger here is due to a deprecated pkg in the dependency chain. [as they won't get the "deprecated version" warning]

Reply:
> I can create a subsequent PR for these pkgs.

You can also push to this PR if you want.

> it won't be obvious to most users that the trigger here is due to a deprecated pkg in the dependency chain

I don't like this, @alalazo @becker33 can we make this message more obvious (in a separate PR)?

Reply:
> You can also push to this PR if you want.

pushed. Thanks!